AutoscoperM/AutoscoperMLib/IO.py

```diff
@@ -255,6 +255,58 @@ def writeTFMFile(filename: str, spacing: list[float], origin: list[float]):
     slicer.mrmlScene.RemoveNode(transformNode)
 
 
+def createTRAFile(
+    volName: str,
+    trialName: str,
+    outputDir: str,
+    trackingsubDir: str,
+    volSize: list[float],
+    Origin2DicomFName: str,
+    transform: vtk.vtkMatrix4x4,
+):
+    transformNode = slicer.vtkMRMLLinearTransformNode()
+    transformNode.SetMatrixTransformToParent(transform)
+    slicer.mrmlScene.AddNode(transformNode)
+
+    if Origin2DicomFName is not None:
+        o2dNode = slicer.util.loadNodeFromFile(Origin2DicomFName)
+        o2dNode.Inverse()
+        transformNode.SetAndObserveTransformNodeID(o2dNode.GetID())
+        transformNode.HardenTransform()
+        slicer.mrmlScene.RemoveNode(o2dNode)
+
+    filename = f"{trialName}_{volName}.tra" if trialName is not None else f"{volName}.tra"
+    filename = os.path.join(outputDir, trackingsubDir, filename)
+
+    if not os.path.exists(os.path.join(outputDir, trackingsubDir)):
+        os.mkdir(os.path.join(outputDir, trackingsubDir))
+
+    tfmMat = vtk.vtkMatrix4x4()
+    transformNode.GetMatrixTransformToParent(tfmMat)
+
+    writeTRA(filename, volSize, tfmMat)
+
+    slicer.mrmlScene.RemoveNode(transformNode)
+    pass
+
+
+def writeTRA(filename: str, volSize: list[float], transform: vtk.vtkMatrix4x4):
+    # Slicer 2 Autoscoper Transform
```

Comment:
Be more descriptive (e.g RAS to LPS ?)

Also can we include a reference to Autoscoper source code and/or documentation?

Reply:
I am not sure that this is a transformation between two conventional coordinate systems such as RAS and LPS, but rather I believe this is a difference in how Slicer and Autoscoper load the data internally. This transformation is something that I determined by visually comparing and manipulating a sample volume in Slicer and Autoscoper.

I opened ISSUE to document this within the RTD. Below I also included the relevant source code from Autoscoper, however, I think this may come down to how LIBTIFF indexes tiff slices and how it reads in an individual slice.

https://github.com/BrownBiomechanics/Autoscoper/blob/8fe9748ab0680955a8c18015009eff0cb457e29a/libautoscoper/src/Volume.cpp#L107-L139

https://github.com/BrownBiomechanics/Autoscoper/blob/8fe9748ab0680955a8c18015009eff0cb457e29a/libautoscoper/src/TiffImage.cpp#L115-L162